pkg/apis/serving/v1alpha1/service_defaults_test.go

```diff
@@ -755,3 +758,108 @@ func TestServiceDefaulting(t *testing.T) {
 		})
 	}
 }
+
+func TestAnnotateUserInfo(t *testing.T) {
+	const (
+		u1 = "oveja@knative.dev"
+		u2 = "cabra@knative.dev"
+		u3 = "vaca@knative.dev"
+	)
+
+	withUserAnns := func(u1, u2 string, s *Service) *Service {
+		a := s.GetAnnotations()
+		if a == nil {
+			a = map[string]string{}
+			s.SetAnnotations(a)
+		}
+		a[serving.CreatorAnnotation] = u1
+		a[serving.UpdaterAnnotation] = u2
+		return s
+	}
+
+	tests := []struct {
+		name     string
+		user     string
+		this     *Service
+		prev     *Service
+		wantAnns map[string]string
+	}{{
+		name: "create-new",
+		user: u1,
+		this: &Service{},
+		prev: nil,
+		wantAnns: map[string]string{
+			serving.CreatorAnnotation: u1,
+			serving.UpdaterAnnotation: u1,
+		},
+	}, {
+		// Old objects don't have the annotation, and unless there's a change in
+		// data they won't get it.
+		name:     "update-no-diff-old-object",
+		user:     u1,
+		this:     &Service{},
+		prev:     &Service{},
+		wantAnns: map[string]string{},
+	}, {
+		name: "update-no-diff-new-object",
+		user: u2,
+		this: withUserAnns(u1, u1, &Service{}),
+		prev: withUserAnns(u1, u1, &Service{}),
+		wantAnns: map[string]string{
+			serving.CreatorAnnotation: u1,
+			serving.UpdaterAnnotation: u1,
+		},
+	}, {
+		name: "update-diff-old-object",
+		user: u2,
+		this: &Service{
+			Spec: ServiceSpec{
+				DeprecatedRelease: &ReleaseType{},
+			},
+		},
+		prev: &Service{
+			Spec: ServiceSpec{
+				DeprecatedRunLatest: &RunLatestType{},
+			},
+		},
+		wantAnns: map[string]string{
+			serving.UpdaterAnnotation: u2,
+		},
+	}, {
+		name: "update-diff-new-object",
+		user: u3,
+		this: withUserAnns(u1, u2, &Service{
```

Comment:
Here user `u3` modified spec from `RunLatestType` to `ReleaseType`
```
		this: withUserAnns(u1, u2, &Service{
			Spec: ServiceSpec{
				DeprecatedRelease: &ReleaseType{},
			},
		}),
		prev: withUserAnns(u1, u2, &Service{
			Spec: ServiceSpec{
				DeprecatedRunLatest: &RunLatestType{},
			},
		}),
```
So annotation `/lastModifier` updates to `u3` 
Because there is a spec change and spec change done by user `u3` so `/lastModifier` updates to `u3`

Reply:
I see. Thanks for the clarification!